core/runtime/src/main/java/io/quarkus/artemis/core/runtime/health/ServerLocatorHealthCheck.java

```diff
@@ -14,34 +14,36 @@
 import org.eclipse.microprofile.health.Readiness;
 
 import io.quarkus.arc.Arc;
+import io.quarkus.artemis.core.runtime.ArtemisRuntimeConfig;
+import io.quarkus.artemis.core.runtime.ArtemisRuntimeConfigs;
 import io.smallrye.common.annotation.Identifier;
 
 @Readiness
 @ApplicationScoped
 public class ServerLocatorHealthCheck implements HealthCheck {
     private final Map<String, ServerLocator> serverLocators = new HashMap<>();
 
-    public ServerLocatorHealthCheck(@SuppressWarnings("CdiInjectionPointsInspection") ArtemisHealthSupport support) {
+    public ServerLocatorHealthCheck(
+            @SuppressWarnings("CdiInjectionPointsInspection") ArtemisRuntimeConfigs runtimeConfigs,
+            @SuppressWarnings("CdiInjectionPointsInspection") ArtemisHealthSupport support) {
         Set<String> names = support.getConfiguredNames();
         Set<String> excludedNames = support.getExcludedNames();
         for (String name : names) {
-            ServerLocator locator = Arc.container().instance(ServerLocator.class, Identifier.Literal.of(name)).get();
-            if (!excludedNames.contains(name) && locator != null) {
-                serverLocators.put(name, locator);
+            if (runtimeConfigs.getAllConfigs().getOrDefault(name, new ArtemisRuntimeConfig()).getUrl() != null) {
+                ServerLocator locator = Arc.container().instance(ServerLocator.class, Identifier.Literal.of(name)).get();
+                if (!excludedNames.contains(name) && locator != null) {
```

Comment:
Why find the locator if it would be excluded?

Reply:
fixed.